File: services/23-laravel/tests/Feature/HealthTest.php
<?php

namespace Tests\Feature;

use Tests\TestCase;

class HealthTest extends TestCase
{
    public function test_hello(): void
    {
        $r = $this->getJson('/');
        $r->assertStatus(200)->assertJsonFragment(['framework' => '23-laravel']);
    }

    public function test_health(): void
    {
        $r = $this->getJson('/health');
        $r->assertStatus(200)->assertJsonFragment(['status' => 'ok']);
    }

    public function test_liveness(): void
    {
        $this->getJson('/health/live')->assertStatus(200);
    }

    public function test_readiness(): void
    {
        $this->getJson('/health/ready')->assertStatus(200);
    }
}

File: services/18-ktor/src/test/kotlin/com/example/HealthTest.kt
package com.example

import io.ktor.client.request.*
import io.ktor.client.statement.*
import io.ktor.http.*
import io.ktor.server.testing.*
import kotlin.test.*

class HealthTest {
    @Test fun testHello() = testApplication {
        application { module() }
        val r = client.get("/")
        assertEquals(HttpStatusCode.OK, r.status)
        assertTrue(r.bodyAsText().contains("Ktor"))
    }

    @Test fun testHealth() = testApplication {
        application { module() }
        val r = client.get("/health")
        assertEquals(HttpStatusCode.OK, r.status)
        assertTrue(r.bodyAsText().contains("ok"))
    }

    @Test fun testLiveness() = testApplication {
        application { module() }
        assertEquals(HttpStatusCode.OK, client.get("/health/live").status)
    }

    @Test fun testReadiness() = testApplication {
        application { module() }
        assertEquals(HttpStatusCode.OK, client.get("/health/ready").status)
    }
}
